document/json/encoder.go

```diff
@@ -20,10 +20,6 @@ type Encoder struct {
 
 // Encode returns the JSON encoding of v.
 func (e *Encoder) Encode(v interface{}) ([]byte, error) {
-	if document.IsNoSerde(v) {
-		return nil, fmt.Errorf("unsupported type: %v", v)
-	}
-
```

Comment:
Wondering if this check should stay? Otherwise I think lines 71-78 in the `encode` function could still serialize the zero value of a generated Smithy type as an empty object which would be weird?

Reply:
in my previous implementation, the `document.IsNoSerde` func was still being called, i had just wrapped it in another function `isUnsupported`. however, in my latest commit, I replaced that to more closely mimic decode which actually directly calls `document.IsNoSerde` when decoding a struct